src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -147,12 +147,18 @@ public Optional<Result> processNewProgrammingExerciseResult(@NotNull Programming
 
             if (buildResult.hasLogs()) {
                 var programmingLanguage = participation.getProgrammingExercise().getProgrammingLanguage();
+                var projectType = participation.getProgrammingExercise().getProjectType();
                 var buildLogs = buildResult.extractBuildLogs(programmingLanguage);
-                buildLogs = buildLogService.removeUnnecessaryLogsForProgrammingLanguage(buildLogs, programmingLanguage);
-                var savedBuildLogs = buildLogService.saveBuildLogs(buildLogs, latestSubmission);
 
-                // Set the received logs in order to avoid duplicate entries (this removes existing logs)
-                latestSubmission.setBuildLogEntries(savedBuildLogs);
+                continuousIntegrationService.get().extractAndPersistBuildLogStatistics(latestSubmission, programmingLanguage, projectType, buildLogs);
+
+                if (!buildResult.isBuildSuccessful()) {
```

Comment:
Why do you only remove the unnecessary logs when the build failed?

Reply:
We only want to store logs in the database, in case the build failed: If the build was successful (it could be executed - not all tests have to pass), the test cases provide better feedback than the log. So we only want to show the build logs as a "fallback" in case no tests are available.

We only remove unnecessary logs, if the build failed as we only save the logs if the build failed. If the build was successful, we do not want to store the logs at all.